subsys/bluetooth/host/conn.c

```diff
@@ -774,7 +774,12 @@ static int conn_prepare_events(struct bt_conn *conn,
 
 	LOG_DBG("Adding conn %p to poll list", conn);
 
-	bool buffers_available = k_sem_count_get(bt_conn_get_pkts(conn)) > 0;
+	/* ISO Synchronized Receiver only builds do not transmit and hence
+	 * may not have any tx buffers allocated in a Controller.
+	 */
+	struct k_sem *conn_pkts = bt_conn_get_pkts(conn);
+	bool buffers_available = conn_pkts ? (k_sem_count_get(conn_pkts) > 0) :
+					     false;
```

Comment:
Just to make sure I fully understand, and to clarify, `bt_conn_get_pkts` returns `NULL` if there are no buffers, instead of returning a `k_sem` with `count == 0`, right? 

Since the comment is explicitly naming the ISO sync receiver, should this "special" (although arguably completely reasonable) handling be guarded with `BT_ISO_SYNC_RECEIVER`?

Reply:
`bt_conn_get_pkts()` can return `NULL` or an uninitialized `&bt_dev.le.acl_pkts` when `bt_dev.le.acl_mtu == 0` (later unlikely in practice), former is what I observed in iso_receive sample (nrf52_bsim segmentation fault in the kernel code).

I am not an expert on the use of `bt_conn_get_pkts()` other places, supplied to `k_sem_take` and `k_sem_give` with return value not checked for `NULL` even though the function `bt_conn_get_pkts()` had NULL as one of its return value.

Someone from host should take over and fix if my quick fix is not acceptable currently.

Reply:
Good point. `bt_conn_get_pkts` can indeed return `NULL`, and isn't NULL-checked anywhere. 

Your check is OK as is, but I guess we should either fix it all placed, or create a GH issue for it